metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/prometheus/PrometheusModel.java

```diff
@@ -45,4 +53,31 @@ public String serialize() {
         return writer.toString();
     }
 
+    private Collector.MetricFamilySamples createMetricFamily(MetricId metricId) {
+        List<Collector.MetricFamilySamples.Sample> sampleList = new ArrayList<>();
+        var samples = new Collector.MetricFamilySamples(metricId.getIdForPrometheus(), Collector.Type.UNKNOWN, "", sampleList);
+        packetsByServiceId.forEach(((serviceId, packets) -> {
+            var serviceName = serviceId.getIdForPrometheus();
+            for (var packet : packets) {
+                Long timeStamp = packet.timestamp * 1000;
+                var dimensions = packet.dimensions();
+                List<String> labels = new ArrayList<>(dimensions.size());
+                List<String> labelValues = new ArrayList<>(dimensions.size());
+                for (var entry : dimensions.entrySet()) {
+                    var labelName = entry.getKey().getIdForPrometheus();
+                    labels.add(labelName);
+                    labelValues.add(entry.getValue());
+                }
+                labels.add("vespa_service");
+                labelValues.add(serviceName);
+
+                Number metric = packet.metrics().get(metricId);
+                if (metric != null) {
+                    sampleList.add(new Collector.MetricFamilySamples.Sample(metricId.getIdForPrometheus(), labels, labelValues, metric.doubleValue(), timeStamp));
+                }
```

Comment:
Can the code skip aggregating values into `labels` and `labelValues` if `metric` is null?

Reply:
Thanks, yes it can, fixed